flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaConnectorOptions.java

```diff
@@ -262,5 +250,81 @@
         EXCEPT_KEY
     }
 
+    /** Startup mode for the Kafka consumer, see {@link #SCAN_STARTUP_MODE}. */
+    public enum ScanStartupMode implements DescribedEnum {
+        EARLIEST_OFFSET("earliest-offset", text("Start from the earliest offset possible.")),
+        LATEST_OFFSET("latest-offset", text("Start from the latest offset.")),
+        GROUP_OFFSETS(
+                "group-offsets",
+                text(
+                        "Start from committed offsets in ZooKeeper / Kafka brokers of a specific consumer group.")),
+        TIMESTAMP("timestamp", text("Start from user-supplied timestamp for each partition.")),
+        SPECIFIC_OFFSETS(
+                "specific-offsets",
+                text("Start from user-supplied specific offsets for each partition."));
+
+        private final String value;
+        private final InlineElement description;
+
+        ScanStartupMode(String value, InlineElement description) {
+            this.value = value;
+            this.description = description;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+
+        @Override
+        public InlineElement getDescription() {
+            return description;
+        }
+    }
+
+    /** Sink semantic, see {@link #SINK_SEMANTIC}. */
+    public enum KafkaSinkSemantic implements DescribedEnum {
+        EXACTLY_ONCE(
+                FlinkKafkaProducer.Semantic.EXACTLY_ONCE,
+                "exactly-once",
+                text(
+                        "Writes all messages in a Kafka transaction that will be committed to Kafka on a checkpoint.")),
+        AT_LEAST_ONCE(
+                FlinkKafkaProducer.Semantic.AT_LEAST_ONCE,
+                "at-least-once",
+                text(
+                        "Waits for all outstanding messages in the kafka buffers to be acknowledged by the Kafka producer on a checkpoint.")),
+        NONE(
+                FlinkKafkaProducer.Semantic.NONE,
+                "none",
+                text(
+                        "No guarantees are made. Messages can be lost and/or duplicated in case of failure."));
+
+        private final FlinkKafkaProducer.Semantic semantic;
+        private final String value;
+        private final InlineElement description;
+
+        KafkaSinkSemantic(
+                FlinkKafkaProducer.Semantic semantic, String value, InlineElement description) {
+            this.semantic = semantic;
+            this.value = value;
+            this.description = description;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+
+        public FlinkKafkaProducer.Semantic getSemantic() {
```

Comment:
default scope (is that allowed?) to not expose it to Table API users, otherwise we should keep a mapping table in the options util